### PR TITLE
[Neo Core UT] Fix ut and style

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

Our UT system is using the snapshot incorrectly, though there isn't any problem in UTs, but they are actually misusing the snapshotcache, and can cause problem if any of the UTs has any operations related to delete value from the cache. 

Fixes # (issue)

## Type of change

- [x] Optimization (the change is only an optimization)
- [ ] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/Neo.Plugins.OracleService.Tests/TestBlockchain.cs [hunk 1]
```diff
@@ -9,28 +9,48 @@
...
 using Neo.Persistence;
 using System;
 
 namespace Neo.Plugins.OracleService.Tests
 {
+
     public static class TestBlockchain
     {
-        public static readonly NeoSystem TheNeoSystem;
+        private static readonly NeoSystem s_theNeoSystem;
```
Comment (Jim8y) [inline tests/Neo.Plugins.OracleService.Tests/TestBlockchain.cs:22]: this should be fine sinse this is UT, not the core

File: tests/Neo.UnitTests/IO/Caching/UT_DataCache.cs [hunk 2]
```diff
@@ -122,12 +122,18 @@ public void TestCommit()
...
+            // If we use myDataCache after it is committed, it will return wrong result.
             myDataCache.Commit();
             Assert.AreEqual(0, myDataCache.GetChangeSet().Count());
 
             store.TryGet(key1.ToArray()).SequenceEqual(value1.ToArray()).Should().BeTrue();
             store.TryGet(key2.ToArray()).Should().BeNull();
             store.TryGet(key3.ToArray()).SequenceEqual(value4.ToArray()).Should().BeTrue();
+
+            myDataCache.TryGet(key1).Value.ToArray().SequenceEqual(value1.ToArray()).Should().BeTrue();
+            // Though value is deleted from the store, the value can still be gotten from the snapshot cache.
```
Comment (Jim8y) [inline tests/Neo.UnitTests/IO/Caching/UT_DataCache.cs:134]: This demonstrate how using snapshotcache after committing it is wrong.

**Review by shargon — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Comment by Jim8y** [comment]
@shargon @cschuchardt88 @chenzhitong @superboyiii

File: tests/Neo.UnitTests/IO/Caching/UT_DataCache.cs [hunk 3]
```diff
@@ -122,12 +122,18 @@ public void TestCommit()
...
+            myDataCache.TryGet(key2).Should().NotBeNull();
```
Comment (Hecate2) [inline tests/Neo.UnitTests/IO/Caching/UT_DataCache.cs:135]:
```suggestion
            myDataCache.TryGet(key2).Value.ToArray().SequenceEqual(value2.ToArray()).Should().BeTrue();
```

Comment (cschuchardt88) [reply tests/Neo.UnitTests/IO/Caching/UT_DataCache.cs:135]:
```csharp
CollectionAssert.AreEqual(value2.ToArray(), myDataCache.TryGet(key2));
```

**Review by shargon — COMMENTED** [review]
conflicts

**Review by cschuchardt88 — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**